awswrangler/opensearch/_write.py

```diff
@@ -0,0 +1,471 @@
+
+    >>> import awswrangler as wr
+    >>> client = wr.opensearch.connect(host='DOMAIN-ENDPOINT')
+    >>> wr.opensearch.index_json(
+    ...     client=client,
+    ...     path='docs.json',
+    ...     index='sample-index1'
+    ... )
+    """
+    # Loading data from file
+
+    if path.startswith("s3://"):
+        bucket, key = parse_path(path)
+        s3 = boto3_session.client('s3')
+        obj = s3.get_object(Bucket=bucket, Key=key)
+        body = obj['Body'].read()
+        lines = body.splitlines()
+        documents = map(lambda x: json.loads(x), lines)
+    else: # local path
+        documents = _file_line_generator(path, is_json=True)
+    return index_documents(
+        client=client,
+        documents=documents,
+        index=index,
+        doc_type=doc_type,
+        **kwargs
+    )
+
+
+def index_csv(
+    client: Elasticsearch,
+    path: Union[str, Path],
+    index: str,
+    doc_type: Optional[str] = None,
+    pandas_kwargs: Optional[Dict[str, Any]] = {},
+    **kwargs
+) -> Dict[str, Any]:
+    """Index all documents from a CSV file to OpenSearch index.
+
+    Parameters
+    ----------
+    client : Elasticsearch
+        instance of elasticsearch.Elasticsearch to use.
+    path : Union[str, Path]
+        s3 or local path to the CSV file which contains the documents.
+    index : str
+        Name of the index.
+    doc_type : str, optional
+        Name of the document type (only for Elasticsearch versions 5.x and older).
+    pandas_kwargs :
+        Dictionary of arguments forwarded to pandas.read_csv().
+        e.g. pandas_kwargs={'sep': '|', 'na_values': ['null', 'none'], 'skip_blank_lines': True}
+        https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html
+    **kwargs :
+        KEYWORD arguments forwarded to :func:`~awswrangler.opensearch.index_documents`
+        which is used to execute the operation
+
+    Returns
+    -------
+    Dict[str, Any]
+        Response payload
+        https://opensearch.org/docs/opensearch/rest-api/document-apis/bulk/#response.
+
+    Examples
+    --------
+    Writing contents of CSV file
+
+    >>> import awswrangler as wr
+    >>> client = wr.opensearch.connect(host='DOMAIN-ENDPOINT')
+    >>> wr.opensearch.index_csv(
+    ...     client=client,
+    ...     path='docs.csv',
+    ...     index='sample-index1'
+    ... )
+
+    Writing contents of CSV file using pandas_kwargs
+
+    >>> import awswrangler as wr
+    >>> client = wr.opensearch.connect(host='DOMAIN-ENDPOINT')
+    >>> wr.opensearch.index_csv(
+    ...     client=client,
+    ...     path='docs.csv',
+    ...     index='sample-index1',
+    ...     pandas_kwargs={'sep': '|', 'na_values': ['null', 'none'], 'skip_blank_lines': True}
+    ... )
+    """
+    custom_pandas_params = {
+        'skip_blank_lines': True,
+        'na_filter': True # will generate Nan value for empty cells. We remove Nan keys in _df_doc_generator
+    }
+    pandas_kwargs.update(custom_pandas_params)
```

Comment:
This will overwrite use input with our defaults - shouldn't it be the other way around?

Reply:
some pandas params I want to enforce:
`skip_blank_lines=True` is crucial (cannot index empty documents).
I also had `na_filter=True` in order to make sure null fields are not indexed into opensearch (anti-pattern), but since the default value for na_filter is True in [pandas.read_csv](https://github.com/pandas-dev/pandas/blob/v1.3.3/pandas/io/parsers/readers.py#L527) I removed this from my code.

I've added a note about the param enforcement in the embedded docs